File: backend/src/main/java/ecrow/backend/business/abstracts/CityService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.core.utilities.results.Result;
import ecrow.backend.entities.concretes.City;

import java.util.List;

public interface CityService {
    DataResult<List<City>> getAll();
    Result existsById(Integer id);
    Result existsByName(String name);
    DataResult<City> getById(Integer id);
    DataResult<City> getByName(String name);

}

File: frontend/src/services/CityService.js
import axios from "axios";

export default class CityService{

    getAll(){
        return axios.get("http://localhost:8080/api/city/getAll")
    }

    getById(id){
        return axios.get(`http://localhost:8080/api/city/getById?id=${id}`)
    }

    getByName(name){
        return axios.get(`http://localhost:8080/api/city/getByName?name=${name}`)
    }
}